libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/InputDialog.cs

```diff
@@ -568,8 +568,19 @@ private async Task<InputState> RecognizeInputAsync(DialogContext dc, int turnCou
         private async Task<DialogTurnResult> PromptUserAsync(DialogContext dc, InputState state, CancellationToken cancellationToken = default(CancellationToken))
         {
             var prompt = await this.OnRenderPromptAsync(dc, state, cancellationToken).ConfigureAwait(false);
+            CheckPrompt(prompt);
+
             await dc.Context.SendActivityAsync(prompt, cancellationToken).ConfigureAwait(false);
+
             return Dialog.EndOfTurn;
         }
+
+        private void CheckPrompt(IActivity prompt)
```

Comment:
I don't think an ArgumentNullException would help the developer here this is a private method and they don't have control on what's sent here, I would rather move the validation logic into the PromptUserAsync method and throw InvalidOperationException explaining the dev that OnRenderPromptAsync returned a null activity. 

something like:
```
            var prompt = await this.OnRenderPromptAsync(dc, state, cancellationToken).ConfigureAwait(false);
            if (prompt == null)
            {
                throw new InvalidOperationException("Call to OnRenderPromptAsync returned a null activity.");
            }

            await dc.Context.SendActivityAsync(prompt, cancellationToken).ConfigureAwait(false);
            return Dialog.EndOfTurn;
```

```